app/src/main/java/org/kiwix/kiwixmobile/webserver/wifi_hotspot/HotspotNotificationManager.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.webserver.wifi_hotspot
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.core.utils.HOTSPOT_SERVICE_CHANNEL_ID
+import org.kiwix.kiwixmobile.webserver.ZimHostActivity
+import javax.inject.Inject
+
+class HotspotNotificationManager @Inject constructor(
+  private val notificationManager: NotificationManager,
+  private val context: Context
+) {
+
+  private fun hotspotNotificationChannel() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      val notificationChannel = NotificationChannel(
+        HOTSPOT_SERVICE_CHANNEL_ID,
+        context.getString(R.string.hotspot_service_channel_name),
+        NotificationManager.IMPORTANCE_DEFAULT
+      )
+      notificationChannel.description = context.getString(R.string.hotspot_channel_description)
+      notificationChannel.setSound(null, null)
+      notificationManager.createNotificationChannel(notificationChannel)
```

Comment:
@gouri-panda the apply was fine, I was just talking about the 1 line `notificationManager.createNotificationChannel(this)`

rewrite it to

```
notificationManager.createNotificationChannel(
  NotificationChannel(...).apply{
    ...
  }
)
```

Reply:
ooh! I get it now the understanding part.

Reply:
without the `val notificationChannel` declaration, exactly like my example. If you name a local variable the exact same as the type of it then you should ask yourself "does this help anyone understand this or am I just making them read the same thing multiple times"

Reply:
>notificationManager.createNotificationChannel(
  NotificationChannel(...).apply{
    ...
  }
)
This looks more readable. so that we don't track anymore.Thank you